packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -81,6 +86,10 @@ class ExpansionPanel {
   /// The widget builder that builds the expansion panels' header.
   final ExpansionPanelHeaderBuilder headerBuilder;
 
+  /// The widget builder that builds the expansion panels' indicator.
+  /// If null, use [ExpandIcon] for default
+  final ExpansionPanelIndicatorBuilder expansionIndicator;
```

Comment:
Could you explain why you would be using a Builder here instead of passing in a Widget directly? I feel like passing a widget directly might be easier.

Reply:
If we passing a widget directyl, then this widget should have one parameter 'isExpanded' ;
To make this works, we should define a base widget at first: 'ExpansionIndicator' with the constructor: ExpansionIndicator(isExpanded:bool);
Then, all the sub-indicators must be derived from this widget;
This is not corresponding to the principle: 'Composition > Inheritance'

Reply:
That makes sense. It could be nice to add a code snippet to the `expansionIndicator` API docs to suggest examples to help other developers understand how to use this property correctly.

Reply:
Sample codes were added to 'ExpansionPanelList'  but not to 'ExpansionPanel'; 
Because there are already sample codes which were added to show the usage of  ExpansionPanelList; Is that ok?